src/main/java/com/r307/arbitrader/service/TradingService.java

```diff
@@ -344,30 +340,24 @@ Integer computeVolumeScale(Exchange exchange, CurrencyPair currencyPair) {
      *
      * @param exchange The exchange to fetch metadata from.
      * @param currencyPair The currency pair to look for.
+     * @param defaultScale If the scale cannot be found, this value will be returned.
      * @return The number of decimals allowed for a price in this currency pair on this exchange.
      */
-    Integer computePriceScale(Exchange exchange, CurrencyPair currencyPair) {
-        final CurrencyPairMetaData defaultPairMetaData = new CurrencyPairMetaData(
-            BigDecimal.ZERO,
-            BigDecimal.ZERO,
-            BigDecimal.valueOf(Long.MAX_VALUE),
-            BTC_SCALE,
-            BTC_SCALE,
-            new FeeTier[0],
-            Currency.BTC
-        );
+    Integer computePriceScale(Exchange exchange, CurrencyPair currencyPair, Integer defaultScale) {
         final ExchangeMetaData exchangeMetaData = exchange.getExchangeMetaData();
-        final CurrencyPairMetaData currencyPairMetaData = exchangeMetaData.getCurrencyPairs().getOrDefault(
-            currencyPair,
-            defaultPairMetaData
-        );
-
-        if (currencyPairMetaData.getPriceScale() == null) {
-            LOGGER.debug("Defaulting to scale of {} for price because metadata is unavailable", USD_SCALE);
-            return USD_SCALE;
+        final Optional<CurrencyPairMetaData> currencyPairMetaDataOptional = Optional.ofNullable(exchangeMetaData.getCurrencyPairs().get(currencyPair));
+
+        if (currencyPairMetaDataOptional.isPresent()) {
+            CurrencyPairMetaData currencyPairMetaData = currencyPairMetaDataOptional.get();
+
+            if (currencyPairMetaData.getPriceScale() != null) {
+                LOGGER.debug("{} {} price scale found in metadata: {}", exchange.getExchangeSpecification().getExchangeName(), currencyPair, currencyPairMetaData.getPriceScale());
+                return currencyPairMetaData.getPriceScale();
+            }
         }
 
-        return currencyPairMetaData.getPriceScale();
+        LOGGER.warn("{} {} price scale unavailable from exchange metadata, using default: {}", exchange.getExchangeSpecification().getExchangeName(), currencyPair, defaultScale);
+        return defaultScale;
```

Comment:
Same question as above:
What is the advantage of changing the scale from 8 (BTC_SCALE) to 1 (the defaultScale we pass in now)?